math/fuzz_test.go

```diff
@@ -22,3 +25,114 @@ func FuzzLegacyNewDecFromStr(f *testing.F) {
 		}
 	})
 }
+
+var (
+	max5Percent, _ = LegacyNewDecFromStr("5") // 5% max tolerance of difference in values
+	decDiv100, _   = LegacyNewDecFromStr("0.01")
+)
+
+func FuzzLegacyDecApproxRoot(f *testing.F) {
+	if testing.Short() {
+		f.Skip("running in -short mode")
+	}
+
+	// 1. Add the corpus: <LEGACY_DEC>,<ROOT>
+	f.Add("-1000000000.5,5")
+	f.Add("1000000000.5,5")
+	f.Add("128,8")
+	f.Add("-128,8")
+	f.Add("100000,2")
+	f.Add("-100000,2")
+
+	// 2. Now fuzz it.
+	f.Fuzz(func(t *testing.T, input string) {
+		splits := strings.Split(input, ",")
+		if len(splits) < 2 {
+			// Invalid input, just skip over it.
+			return
+		}
```

Comment:
Many input strings are going to be immediate rejected. I suggest tightening the fuzzing function to something like


```go
func(t *testing.T, dec string, nthRoot uint64)
```

Further, I think you can avoid invalid strings by going through `big.Rat`:

```go
func(t *testing.T, nomBytes, denomBytes []byte, nthRoot uint64)
   nom := big.Int{}.SetBytes(nomBytes)
   denom := big.Int{}.SetBytes(denomBytes)
  decRat := big.Rat{}.SetFrac(nom, denom)
  dec := big.Float{}.SetRat(decRat)
```

If you want negative numbers, add a `bool` argument to the fuzzing function.